### ISSUE TITLE
azurerm_kubernetes_cluster forces redeployment of recent cluster due to windows_profile

### ISSUE THREAD

**Issue by EPinci** [terraform-providers/terraform-provider-azurerm#6235]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

TF v0.12.24 - AzureRM 2.2.0

### Affected Resource(s)

azure_kubernetes_cluster

### Terraform Configuration Files

Any recently deployed cluster with no "windows_profile" gets provisioned fine.
On any subsequent run, TF tries to redeploy the cluster referencing a change in that block.

```
      - windows_profile {
          - admin_username = "azureuser" -> null # forces replacement
        }
```

### References

* See behavior change in Azure AKS at https://github.com/Azure/AKS/releases/tag/2020-03-09

**Comment by baosen** [comment]
A simple workaround for this issue can be made by adding this small code snippet to the resource:
```
...
 lifecycle {
    ignore_changes = [
      windows_profile,
    ]
  }
...
```

**Comment by WodansSon** [comment]
@EPinci, thanks for opening this issue... I am currently unable to repro this issue on my local dev machine. The current resource is using the 2019-11-01 version of the API and the change mentioned above wasn't introduced until the 2020- 03-09 version of the API. Are you by any chance happen to be running a private version of the provider with newer AKS API's?

**Comment by Cyanopus** [comment]
This behavior can be seen in AzureRM 2,3 as well. Its not affecting previously created resources before Azure done this change on the API. Which means that if you've provisioned the cluster a week ago you are most likely safe in the resource.

**Comment by runecalico** [comment]
So far it looks like another option for those who aren't entirely happy with adding an ignore_changes to their code (say if they have a azurerm_kubernetes_cluster) is to just set the windows_profile block as it looks like at the moment you can set those values even if you don't have any windows nodes ..

**Comment by tscully49** [comment]
Is this going to be patched for the azurerm provider 1.44.0?  My company uses that version and we are running into the same issue.

**Comment by Cyanopus** [comment]
@tscully49  You will probably have to upgrade to 2.5.0
